h2o-core/src/main/java/water/fvec/Frame.java

```diff
@@ -1665,7 +1702,40 @@ public int getCurrentRowSize() throws IOException {
       return StringUtils.bytesOf(sb);
     }
 
-    @Override public int available() throws IOException {
+    /**
+     * Returns a factor escaped for CSV format, if required.
+     *
+     * @param vectorId              Id of the categorical vector
+     * @param categoricalLevelIndex Index of the categorical level in the original domain
+     * @param vec                   Vector with it's domain
+     * @return A {@link String} representation of the categorical level, with double-quotes escaped for CSV.
+     */
+    private String getEscapedCategoricalLevel(final int vectorId, final int categoricalLevelIndex, final Vec vec) {
+      String[] domain = escapedCategoricalLevels.get(vectorId);
```

Comment:
would rename this to escapedDomain to make it more readable

Reply:
Yeah, that's nasty. Corrected.